### PR TITLE
Introduce Pytest

### PR DESCRIPTION
**Pull request by jamesvandyne**
This PR introduces the [pytest](https://pytest.org) library for running / writing tests. Pytest seems to be the modern standard for testing with Python. As the library expands in functionality going forward, pytest will allow us to easily split our tests in modules making it easier to maintain.

Also ss part of the introduction, I need to move the indieweb-utils file to the `src/` directory and tests to the `tests/` as is best practice for pytest. The pytest docs link to a great blog post about the topic [Packaging a python library](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure).

### CODE REVIEW

File: tests/test_indieweb_utils.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Partial test coverage is implemented for various functions in the library
+# Assistance is requested to achieve full test coverage
+import pytest
+
+from indieweb_utils import *
+
+
+@pytest.mark.parametrize(
```
Comment (jamesvandyne) [inline tests/test_indieweb_utils.py:8]: Use pytest parametrize so each combination is treated as its own test, rather the one big test.

Comment (capjamesg) [reply tests/test_indieweb_utils.py:8]: Sounds good, thanks!

**Comment by capjamesg** [comment]
I will merge the pytest branch into master after reviewing all of your changes. Per [this commit](https://github.com/capjamesg/indieweb-utils/pull/4/commits/2d4ff809e026c11eb09ce221e811fbd7fdc3fc36), is it worth also adding a note that mentions test coverage is a work in progress?

**Comment by jamesvandyne** [comment]
@capjamesg I don't think we need to add a note about tests being wip, as they (usually) change whenever the library changes. They are forever wip 😀

**Pull request merged**